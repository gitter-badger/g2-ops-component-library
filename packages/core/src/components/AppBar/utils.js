import { compose } from 'ramda'

const FIND_ALL_UNDERSCORES = /_+/g
const FIND_FIRST_LETTER_REGEX = /\w\S*/g

export const replaceUnderscoreWithSpace = (string) =>
  string && string.length > 0 && string.replace(FIND_ALL_UNDERSCORES, ' ')

export const capitalizeFirstLetter = (string) =>
  string &&
  string.replace(
    FIND_FIRST_LETTER_REGEX,
    (match) => match && `${match.charAt(0).toUpperCase()}${match.substr(1).toLowerCase()}`
  )

export const beautifyRoleText = compose(capitalizeFirstLetter, replaceUnderscoreWithSpace)