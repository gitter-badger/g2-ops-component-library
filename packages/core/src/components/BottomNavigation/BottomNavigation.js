// @flow
import type { Node, ChildrenArray } from 'react'

import React from 'react'
import { BottomNavigation as MuiBottomNavigation, BottomNavigationItem } from 'material-ui/BottomNavigation'

import { wrapMuiContext } from 'utilities/wrapMuiContext'

export const BottomNavigation = wrapMuiContext(<T: { children?: ChildrenArray<Node> }>(props: T) => (
  <MuiBottomNavigation {...props}>{props.children}</MuiBottomNavigation>
))

// export default wrapMuiContext(BottomNavigation)
export { BottomNavigationItem }
