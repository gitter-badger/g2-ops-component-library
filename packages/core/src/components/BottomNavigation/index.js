// export { BottomNavigation, BottomNavigationItem } from './BottomNavigation'
export { BottomNavigation, BottomNavigationItem } from './BottomNavigation'
