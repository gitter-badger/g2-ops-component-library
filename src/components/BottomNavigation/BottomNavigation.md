Bottom Navigation Example:

```js
var BottomNavExample = require('../../examples/BottomNavExample.js').default;
<BottomNavExample />
```