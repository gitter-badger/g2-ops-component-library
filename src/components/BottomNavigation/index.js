export { BottomNavigation, BottomNavigationItem } from './BottomNavigation'
export default from './BottomNavigation'
