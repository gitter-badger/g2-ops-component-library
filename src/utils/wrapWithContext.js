import React from 'react'
import WithContext from 'react-with-context'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import createMuiTheme from 'material-ui/styles/getMuiTheme'

const wrapWithContext = (component, context) => <WithContext context={context}>{component}</WithContext>

export const theme = createMuiTheme(baseTheme)

export const wrapWithMaterialUIContext = (component) =>
  wrapWithContext(component, { muiTheme: createMuiTheme(baseTheme) })

export default wrapWithContext
