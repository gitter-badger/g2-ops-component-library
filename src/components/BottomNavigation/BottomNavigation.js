import React from 'react'
import { BottomNavigation as MuiBottomNavigation, BottomNavigationItem } from 'material-ui/BottomNavigation'
import { wrapMuiContext } from '../../wrapMuiContext'

const BottomNavigation = (props) => (
  <MuiBottomNavigation {...props}>
    {props.children}
  </MuiBottomNavigation>
)

export default wrapMuiContext(BottomNavigation)
export { BottomNavigationItem }
