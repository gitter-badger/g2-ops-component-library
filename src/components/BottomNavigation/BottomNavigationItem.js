import { BottomNavigationItem } from 'material-ui/BottomNavigation'
export default BottomNavigationItem