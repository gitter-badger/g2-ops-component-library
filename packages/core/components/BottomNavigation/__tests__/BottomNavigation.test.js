import React from 'react'
import renderer from 'react-test-renderer'

import { wrapWithMaterialUIContext } from 'utils/wrapWithContext'

import BottomNavigation from '../BottomNavigation'

// Nothing added here as it is just wrapper around MaterialBottomNavigation
describe('<BottomNavigation />', () => {
  test('should render properly with no children', () => {
    const tree = renderer.create(wrapWithMaterialUIContext(<BottomNavigation />))
    expect(tree).toMatchSnapshot()
  })

  test('should render properly with children', () => {
    const tree = renderer.create(
      wrapWithMaterialUIContext(
        <BottomNavigation>
          <div className="bottomNavigationChildren" />
        </BottomNavigation>,
      ),
    )
    expect(tree).toMatchSnapshot()
  })
})
