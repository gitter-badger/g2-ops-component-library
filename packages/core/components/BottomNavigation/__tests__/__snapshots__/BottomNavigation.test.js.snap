// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomNavigation /> should render properly with children 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "height": 56,
      "justifyContent": "center",
      "muiPrepared": true,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="bottomNavigationChildren"
    selected={false}
    style={
      Object {
        "flex": "1",
      }
    }
  />
</div>
`;

exports[`<BottomNavigation /> should render properly with no children 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "height": 56,
      "justifyContent": "center",
      "muiPrepared": true,
      "position": "relative",
      "width": "100%",
    }
  }
/>
`;
